aptos-move/framework/aptos-framework/sources/SimpleMap.move

```diff
@@ -0,0 +1,220 @@
+/// This module provides a solution for sorted maps, that is it has the properties that
+/// 1) Keys point to Values
+/// 2) Each Key must be unique
+/// 3) A Key can be found within O(Log N) time
+/// 4) The data is stored as a sorted by Key
+/// 5) Adds and removals take O(N) time
+module AptosFramework::SimpleMap {
+    use Std::Errors;
+    use Std::Option;
+    use Std::Vector;
+    use AptosFramework::Comparator;
+
+    const EKEY_ALREADY_EXISTS: u64 = 0;
+    const EKEY_NOT_FOUND: u64 = 1;
+
+    struct SimpleMap<Key: store, Value: store> has store {
```

Comment:
could we support the copy for the simpleMap? Otherwise, we can just use Table to write a map

Reply:
I think it might be used in an event and then it requires a droppable ability

Reply:
I think we probably should consider moving away from assuming that every piece of data modified on the chain should result in an event. Also if we start putting requirements on types within the map, we limit its utility. Consider the extreme case where an NFT changes off-chain data but wants to notify on-chain.

Reply:
`Events are not guaranteed to be canonical, other devs will most certainly have less comprehensive events and may skip over them entirely. Indexers will need to read the chain to validate data is correct and should only use events as a notification of a state change` totally agree that event is not guaranteed to be the source of truth for non-framework modules.  For non-framework contracts, I guess there will be customized indexer built by the other people that can read writesets. 

For framework modules, if we provide comprehensive events of the framework, would it be easier to build indexer or read the data change on the core modules of our framework?

Reply:
I think that might teach people to blindly believe that events are canonical. I was chatting with several folks in other ecosystems and they emphasized many times how dangerous events can be.